### DISCUSSION TITLE
Question: Role of MCSampler in qExpectedHypervolumeImprovement

### DISCUSSION THREAD

**Discussion by 1bing2** [meta-pytorch/botorch#1281] [Q&A] [answered]
Hello，everyone:
I am new here. I have a question to consult.When I read the botorch tutorials, the  function called  qExpectedHypervolumeImprovement
have a MCSampler ,how can l understand this MCsampler?
or  Can you explain the usage of this or give some information about it?

**Comment by Balandat** [comment, marked as answer]
`qExpectedHypervolumeImprovement` (and more generally other Monte Carlo acquisition functions in BoTorch) uses Monte Carlo sampling from the surrogate model's posterior in order to compute the acquisition function. For a general introduction to this in BoTorch see sections 3+4 of https://proceedings.neurips.cc/paper/2020/file/f5b1b89d98b7286673128a5fb112cb9a-Paper.pdf, for the specifics about how it's used in `qExpectedHypervolumeImprovement`, see https://proceedings.neurips.cc/paper/2020/file/6fec24eac8f18ed793f5eaad3dd7977c-Paper.pdf. 

The `MCSampler` is essentially an abstraction that makes it convenient to draw samples from model posteriors while potentially fixing the base samples for Sample Average Approximation (SAA) - see the linked BoTorch paper for more details.
